torch/_functorch/aot_autograd.py

```diff
@@ -973,6 +1176,16 @@ def inner(*flat_args):
         # intermediate bases are also included in the backward graph
         f_tangents = f_input_tangents + f_output_tangents + intermediate_bases
         traced_tangents = pytree.tree_map(from_fun, f_tangents)
+        user_outs = pytree.tree_map(from_fun, f_output_tangents)
+
+        f_mutated_inputs = [
+            inp
+            for inp, info in zip(flat_f_args, input_info)
+            if info.mutates_data or info.mutates_metadata
+        ]
+        # When handling subclasses, we need info about **all** outputs of compiled forward graph,
+        # so we know precisely which graph outputs to wrap back into tensor subclasses
+        fw_graph_outs = pytree.tree_map(from_fun, f_mutated_inputs + list(flat_f_outs) + intermediate_bases)
```

Comment:
TODO: review this

Reply:
Is this the same as output_info?

Reply:
https://github.com/pytorch/pytorch/pull/104483/files#r1350813081